uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeavePeriodEntitlementTest.php

```diff
@@ -106,4 +110,251 @@ public function testGetBreakdownWhenOnlyEntitlementIdIsPassed() {
     ];
     $this->assertEquals($expectedResult, $result);
   }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage You must include either the id of a specific entitlement, or both the contact and period id
+   */
+  public function testGetEntitlementDoesNotAcceptContactIdWithoutPeriodId() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['contact_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage You must include either the id of a specific entitlement, or both the contact and period id
+   */
+  public function testGetEntitlementDoesNotAcceptPeriodIdWithoutContactId() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['period_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage You must include either the id of a specific entitlement, or both the contact and period id
+   */
+  public function testGetEntitlementWhenAllParametersArePassed() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['entitlement_id'=> 1, 'period_id' => 1, 'contact_id'=>1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage You must include either the id of a specific entitlement, or both the contact and period id
+   */
+  public function testGetEntitlementWhenEntitlementIdAndPeriodIdArePassed() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['entitlement_id'=> 1, 'contact_id'=>1]);
```

Comment:
The description of this test is wrong. It should be `testGetEntitlementWhenEntitlementIdAndContactIdArePassed`.
We should also add an additional one for `testGetEntitlementWhenEntitlementIdAndPeriodIdArePassed`